streaming/src/main/scala/io/gearpump/streaming/executor/Executor.scala

```diff
@@ -122,109 +131,159 @@ class Executor(executorContext: ExecutorContext, userConf : UserConfig, launcher
-      tasks.get(registered.taskId).foreach {
-        case actorRef: ActorRef =>
-          tasks += registered.taskId -> actorRef
-          actorRef forward registered
-      }
-    case rejected: TaskRejected =>
-      tasks.get(rejected.taskId).foreach {
-        case task: ActorRef => task ! PoisonPill
-      }
-      tasks -= rejected.taskId
-      LOG.error(s"Task ${rejected.taskId} is rejected by AppMaster, shutting down it...")
+        context.become(dynamicDagPhase1(dagVersion, launched, changed, registered :+ confirm.taskId))
+
+      case rejected: TaskRejected =>
+        // means this task shoud not exists...
+        tasks.get(rejected.taskId).foreach {
+          case task: ActorRef => task ! PoisonPill
+        }
+        tasks -= rejected.taskId
+        LOG.error(s"Task ${rejected.taskId} is rejected by AppMaster, shutting down it...")
 
-    case register: RegisterTask =>
-      sendMsgWithTimeOutCallBack(appMaster, register, registerTaskTimeout, timeOutHandler(register.taskId))
+      case register: RegisterTask =>
+        appMaster ! register
+    })
   }
 
-  def applicationReady(dagVersion: Int): Receive = executorService orElse {
-    case launch: LaunchTasks =>
-      LOG.info(s"received $launch, Executor transit to dag version: ${launch.dagVersion} from current version $dagVersion")
-      context.become(dynamicDag(launch.dagVersion, List.empty[TaskId], List.empty[ChangeTask]))
-      self forward launch
+  def dynamicDagPhase2(dagVersion: Int, launched: List[TaskId], changed: List[ChangeTask]): Receive = {
+    LOG.info("Transit to dynamic Dag Phase2")
+    state = State.DYNAMIC_DAG_PHASE2
+    box {
+      case startAll@StartAllTasks(version) =>
+        LOG.info(s"Start All Tasks...")
+        assertVersion(dagVersion, version, clue = startAll)
 
-    case change: ChangeTasks =>
-      LOG.info(s"received $change, Executor transit to dag version: ${change.dagVersion} from current version $dagVersion")
-      context.become(dynamicDag(change.dagVersion, List.empty[TaskId], List.empty[ChangeTask]))
-      self forward change
+        launched.foreach(taskId => tasks.get(taskId).foreach(_ ! StartTask(taskId)))
+        changed.foreach(changeTask => tasks.get(changeTask.taskId).foreach(_ ! changeTask))
+
+        taskArgumentStore.removeNewerVersion(dagVersion)
+        taskArgumentStore.removeObsoleteVersion
+        context.become(applicationReady(dagVersion))
+    }
   }
 
-  private val timeOutHandler = (taskId: TaskId) => {
-    val cause = s"Failed to register task $taskId to AppMaster of application $appId, " +
-      s"executor id is $executorId, treat it as MessageLoss"
-    LOG.error(cause)
-    appMaster ! MessageLoss(executorId, taskId, cause)
+  def applicationReady(dagVersion: Int): Receive = {
+    state = State.ACTIVE
+    transitionEnd = System.currentTimeMillis()
+    LOG.info("Transit to state Application Ready. This transition takes " + (transitionEnd - transitionStart) + "milliseconds")
```

Comment:
should not print this on start up. Otherwise, the following log doesn't make sense.

```
Transit to state Application Ready. This transition takes 1450074184253milliseconds
```

Also, please add a space before "milliseconds"